product-service/src/lambda/get-product-by-id.ts

```diff
@@ -0,0 +1,29 @@
+import { productService } from "../services/product-service";
+import { lambda } from "../libs/lambda";
+import { httpResponse } from "../libs/http";
+
+type Params = {
+  productId?: string;
+};
+
+const isInvalidParams = (p: Params) => !p.hasOwnProperty("productId");
+
+export const getProductById = lambda(async (event) => {
+  const params = Object(event.pathParameters);
+
+  if (isInvalidParams(params)) {
+    return httpResponse.failureResult({ name: "validation error" });
+  }
+
+  const productId = params["productId"];
+
+  const product = await productService.getProductById(productId);
+
+  if (!product) {
+    return httpResponse.failureResult({
+      name: `Product not found by id: ${productId}`,
+    });

```

Comment:
This will return 400 instead of 404
Besides, this will never get executed, because getProductById is returning rows instead of a product. So you will get 200 with an empty array instead of 404

Reply:
ok, work on it

Reply:
done, deployed

Reply:
fix it, deployed